src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,247 @@
+// eslint-disable-next-line
+import { useParams } from 'react-router';
+import { useEffect, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
 
-export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+export const PeoplePage: React.FC<{
+  isError: boolean;
+  setIsError: React.Dispatch<React.SetStateAction<boolean>>;
+  isLoading: boolean;
+  peopleFromServer: Person[] | undefined;
+  errorMessage: string | undefined;
+}> = ({
+  isError, setIsError, isLoading, peopleFromServer, errorMessage,
+}) => {
+  const { slug = '' } = useParams();
+  const [visiblePeople, setVisiblePeople] = useState(peopleFromServer || []);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const initialSort = searchParams.get('sort');
+  const [sortUpdate, SetSortUpdate] = useState(initialSort);
+  const [sortBefore, SetSortBefore] = useState(sortUpdate);
+  const initialOrder = searchParams.get('order') || null;
+  const [orderUpdate, SetOrderUpdate] = useState(initialOrder);
+  const sortLane = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const setSearchWith = (
+    firstSearchParams: URLSearchParams,
+    params: {
+      [key: string]: string[] | string | null;
+    },
+  ) => {
+    Object.entries(params).forEach(([key, value]) => {
+      if (value === null || value === '') {
+        firstSearchParams.delete(key);
+      } else if (Array.isArray(value)) {
+        firstSearchParams.delete(key);
+
+        value.forEach((part) => {
+          firstSearchParams.append(key, part);
+        });
+      } else {
+        firstSearchParams.set(key, value);
+      }
+    });
+
+    return firstSearchParams.toString();
+  };
+
+  const askProperty = (element: string) => {
+    if (orderUpdate) {
+      return {
+        search: setSearchWith(searchParams, {
+          sort: null,
+          order: null,
+        }),
+      };
+    }
+
+    return {
+      search: setSearchWith(searchParams, {
+        sort: element,
+        order: element === sortUpdate ? 'desc' : null,
+      }),
+    };
+  };
+
+  const doSortingNow = () => {
+    if (peopleFromServer) {
+      const resetSex = searchParams.get('sex') === null;
+      const resetCenturies = searchParams.get('centuries') === null;
+      const resetSearch = searchParams.get('search') === null;
+      const resetSort = searchParams.get('sort') === null;
+
+      const sortElement = (currentList: Person[]) => {
+        if (resetSort) {
+          return currentList;
+        }
```

Comment:
pass `resetSort` as an argument and move this function to a separate file

Reply:
I can move out from "doSortingNow", but cant move to a separate file, becouse this function use very much variables, writing in "PeoplePage".